src/librustdoc/html/static/main.js

```diff
@@ -926,40 +917,14 @@ function hideThemeButtonState() {
                 e.innerHTML = labelForToggleButton(false);
             });
             innerToggle.title = "collapse all docs";
-            if (fromAutoCollapse !== true) {
-                onEachLazy(document.getElementsByClassName("collapse-toggle"), function(e) {
-                    collapseDocs(e, "show");
-                });
-            }
         } else {
             addClass(innerToggle, "will-expand");
-            onEachLazy(document.getElementsByTagName("details"), function(e) {
-                e.open = false;
-            });
-            onEveryMatchingChild(innerToggle, "inner", function(e) {
-                var parent = e.parentNode;
-                var superParent = null;
-
-                if (parent) {
-                    superParent = parent.parentNode;
-                }
-                if (!parent || !superParent || superParent.id !== "main" ||
-                    hasClass(parent, "impl") === false) {
-                    e.innerHTML = labelForToggleButton(true);
-                }
-            });
-            innerToggle.title = "expand all docs";
             if (fromAutoCollapse !== true) {
```

Comment:
As far as I can tell, there is nothing that calls `toggleAllDocs(true)`. I suspect this `if` statement should be removed. Or, zooming out a little: toggleAllDocs can now be dramatically simplified. In the "open" case, it should iterate over all `.rustdoc-toggle` and set `open = true`; in the close case it should iterate over all `.rustdoc-toggle` and set `open = false` (and of course it should update the "collapse all docs" button itself).

Reply:
Not all details have the `rustdoc-toggle` class iirc.

Reply:
If that's the case, it's an oversight and we should correct it. Grepping, I see two instances in mod.rs where we emit a `<details>` with no `rustdoc-toggle`. Would you mind including updates to those in this PR?

Reply:
Oh actually it's fine: we didn't expand those items' docs before, so it's actually fine to just go through all "rustdoc-toggle" and ignore the others.